server/src/main/java/org/elasticsearch/action/admin/indices/create/AutoCreateAction.java

```diff
@@ -116,25 +116,25 @@ protected ClusterStateUpdateResponse newResponse(boolean acknowledged) {
 
                 @Override
                 public ClusterState execute(ClusterState currentState) throws Exception {
-                    DataStreamTemplate dataStreamTemplate = resolveAutoCreateDataStream(request, currentState.metadata());
+                    final String indexName = indexNameExpressionResolver.resolveDateMathExpression(request.index());
+
+                    // This will throw an exception if the index or data stream does not exist and creating it is prohibited.
+                    final boolean shouldAutoCreate = autoCreateIndex.shouldAutoCreate(indexName, currentState);
```

Comment:
Moving this call here has the side effect of also applying `action.auto_create_index` to data streams if the template does not have `allow_auto_create` set, which I don't think is what we wanted to do.

Reply:
I must have misunderstood - to clarify, were you expecting that all I was going to do was apply the value of `allow_auto_create` to data streams, as opposed to also making them subject to `auto_create_index`? I'm happy to implement either, it just seemed strange that a user could control auto-creation for indices, but not for data streams. @jaymode what's your feeling here?

Reply:
At this point, making `action.auto_create_index` apply to data streams would be a breaking change, no? With `allow_auto_create`, that's a new feature so we can define the behavior however we want, but we have to be more careful with behavior we've already shipped.

Reply:
Ah, that's a good point. I'll rework the code.

Reply:
I've reworked `AutoCreateAction` - what do you think about the docs for `allow_auto_create`, is it worth discussing the difference between indices and data streams w.r.t the new template setting?

Reply:
I think the fact that `action.auto_create_index` controls whether data streams can be auto created can be seen as a bug. Something that we missed during development. I would prefer that `action.auto_create_index` doesn't affect the auto creation of data streams.

Reply:
Never mind my comment. The PR is doing what I suggested in my comment.